crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1193,6 +1200,14 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                 ruff::rules::never_union(checker, expr);
             }
 
+            // Avoid duplicate checks if the parent is a literal, since this rule already
+            // traverses nested literals.
+            if checker.enabled(Rule::DuplicateLiteralMember) {
+                if !checker.semantic.in_nested_literal() {
+                    flake8_pyi::rules::duplicate_literal_member(checker, expr);
+                }
+            }
+
```

Comment:
Hmm, why are we checking `BinOp` expressions here? A `Union` can either be a `Subscript` expression (`Union[int, str]`, etc.) or a `BitOr` expression (`int | str`, etc.). But AFAIK a `Literal` can only be a `Subscript` expression, right?

Reply:
I'm not sure what you mean by `BinOp` being checked here, this code only checks for `Subscript` exprs as far as I understand.

Reply:
You've added a hook here so that the visitor that traverses the AST calls the `duplicate_literal_member` function on every `BinOp` node that it encounters:

![image](https://github.com/astral-sh/ruff/assets/66076021/fecbaeb1-9019-4d73-b9b9-293d7935ad1a)

Reply:
Oh right. Meant to place it 1 level above.

Reply:
It was completely unnecessary, removed.